sunpy/net/helioviewer.py

```diff
@@ -57,6 +57,14 @@ def data_sources(self):
         # Sort the output for printing purposes
         return OrderedDict(sorted(data_sources_dict.items(), key=lambda x: x[1]))
 
+    def _partial_match(self, key, dictionary):
+        """
+        A function that returns source id based on partial key.
+        """
```

Comment:
The function is more general than that.

Also if you found this online, I would link to it so it adds more context for someone reading the code.

Reply:
I will add some description